platform/darwin/src/NSComparisonPredicate+MGLAdditions.mm

```diff
@@ -1,218 +1,36 @@
-                            format:@"Predicate cannot compare values IN attribute."];
-            }
-            
-            // Convert $type IN to TypeInFilter.
-            if ([leftExpression.keyPath isEqualToString:@"$type"]) {
-                mbgl::style::TypeInFilter typeInFilter;
-                typeInFilter.values = rightExpression.mgl_aggregateFeatureType;
-                return typeInFilter;
-            }
-            
-            // Convert $id IN to IdentifierInFilter.
-            if ([leftExpression.keyPath isEqualToString:@"$id"]) {
-                mbgl::style::IdentifierInFilter idInFilter;
-                idInFilter.values = rightExpression.mgl_aggregateFeatureIdentifier;
-                return idInFilter;
-            }
-            
-            mbgl::style::InFilter inFilter;
-            inFilter.key = leftExpression.keyPath.UTF8String;
-            inFilter.values = rightExpression.mgl_aggregateMBGLValue;
-            return inFilter;
-        }
-        case NSContainsPredicateOperatorType: {
-            if (!isReversed) {
-                if (rightType == NSConstantValueExpressionType && [rightExpression.constantValue isKindOfClass:[NSString class]]) {
-                    [NSException raise:NSInvalidArgumentException
-                                format:@"IN not supported for string comparison."];
-                }
-                [NSException raise:NSInvalidArgumentException
-                            format:@"Predicate cannot compare attribute CONTAINS values."];
-            }
-            
-            // Convert CONTAINS $type to TypeInFilter.
-            if ([rightExpression.keyPath isEqualToString:@"$type"]) {
-                mbgl::style::TypeInFilter typeInFilter;
-                typeInFilter.values = leftExpression.mgl_aggregateFeatureType;
-                return typeInFilter;
-            }
-            
-            // Convert CONTAINS $id to IdentifierInFilter.
-            if ([rightExpression.keyPath isEqualToString:@"$id"]) {
-                mbgl::style::IdentifierInFilter idInFilter;
-                idInFilter.values = leftExpression.mgl_aggregateFeatureIdentifier;
-                return idInFilter;
-            }
-            
-            mbgl::style::InFilter inFilter;
-            inFilter.key = rightExpression.keyPath.UTF8String;
-            inFilter.values = leftExpression.mgl_aggregateMBGLValue;
-            return inFilter;
-        }
+    switch (self.predicateOperatorType) {
+        case NSEqualToPredicateOperatorType:
+        case NSNotEqualToPredicateOperatorType:
+        case NSGreaterThanPredicateOperatorType:
+        case NSGreaterThanOrEqualToPredicateOperatorType:
+        case NSLessThanPredicateOperatorType:
+        case NSLessThanOrEqualToPredicateOperatorType:
+        case NSInPredicateOperatorType:
+        case NSContainsPredicateOperatorType:
         case NSBetweenPredicateOperatorType: {
```

Comment:
We don’t really need a switch statement here anymore. `-mgl_jsonExpressionObject` can be responsible for raising an exception when the operator type is unsupported.